src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1502,23 +1519,108 @@ lemma preconnected.subsingleton_connected_component (h : G.preconnected) :
+  (φ : G →g G') (ψ : G' →g G'') : (C.map φ).map ψ = C.map (ψ.comp φ) :=
 by { refine C.ind _, exact (λ _, rfl), }
 
+variables {φ : G ≃g G'} {v : V} {v' : V'}
+
+@[simp] lemma connected_component.iso_image_comp_eq_map_iff_eq_comp
+  {C : G.connected_component} :
+  (G'.connected_component_mk (φ v)) = C.map φ.to_hom ↔ (G.connected_component_mk v) = C :=
+begin
+  refine C.ind (λ u, _),
+  simp only [connected_component.map_mk, connected_component.eq],
+  exact iso.reachable_iff,
+end
+
+@[simp] lemma connected_component.iso_inv_image_comp_eq_iff_eq_map
+  {C : G.connected_component} :
+    G.connected_component_mk (φ.symm v') = C ↔ G'.connected_component_mk v' = C.map φ :=
+begin
+  refine C.ind (λ u, _),
+  simp only [connected_component.map_mk, connected_component.eq],
+  exact iso.symm_apply_reachable,
+end
+
+/-- An isomorphism of graphs induces a bijection of connected components. -/
+@[simps]
+def connected_component.iso (φ : G ≃g G') : G.connected_component ≃ G'.connected_component :=
+{ to_fun := connected_component.map φ.to_hom,
+  inv_fun := connected_component.map φ.symm.to_hom,
+  left_inv := λ C, connected_component.ind
+    (λ v, congr_arg (G.connected_component_mk) (equiv.left_inv φ.to_equiv v)) C,
+  right_inv := λ C, connected_component.ind
+    (λ v, congr_arg (G'.connected_component_mk) (equiv.right_inv φ.to_equiv v)) C }
+
+/-- The set of vertices in a connected component of a graph. -/
+def connected_component.supp (C : G.connected_component) :=
+  { v | G.connected_component_mk v = C }
+
+@[ext] lemma connected_component.supp_injective :
+  function.injective (connected_component.supp : G.connected_component → set V) :=
+begin
+  refine connected_component.ind₂ _,
+  intros v w h,
+  simp only [set.ext_iff, connected_component.eq, set.mem_set_of_eq, connected_component.supp] at
+     h ⊢,
+  exact ((h v).mp (reachable.refl _)),
+end
+
+@[simp]
+lemma connected_component.supp_inj {C D : G.connected_component} : C.supp = D.supp ↔ C = D :=
+connected_component.supp_injective.eq_iff
+
+instance : set_like G.connected_component V :=
+{ coe := connected_component.supp,
+  coe_injective' := connected_component.supp_injective, }
+
+@[simp] lemma connected_component.mem_supp_iff (C : G.connected_component) (v : V) :
+  v ∈ C.supp ↔ G.connected_component_mk v = C := iff.rfl
+
+lemma connected_component_mk_mem (G : simple_graph V) {v : V} :
+  v ∈ G.connected_component_mk v := by { exact rfl, }
+
+/--
+The equivalence between connected components, induced by an isomorphism of graphs,
+itself defines an equivalence on the supports of each connected component.
+-/
+def connected_component.iso_equiv_supp (φ : G ≃g G') (C : G.connected_component) :
+  C.supp ≃ (connected_component.iso φ C).supp :=
+{ to_fun := λ v, ⟨φ.to_fun v.val, connected_component.iso_image_comp_eq_map_iff_eq_comp.mpr v.prop⟩,
+  inv_fun := λ v', ⟨φ.inv_fun v', connected_component.iso_inv_image_comp_eq_iff_eq_map.mpr v'.prop⟩,
```

Comment:
I haven't tested this, but these should work, and it would be preferred:
```suggestion
{ to_fun := λ v, ⟨φ v, connected_component.iso_image_comp_eq_map_iff_eq_comp.mpr v.prop⟩,
  inv_fun := λ v', ⟨φ.symm v', connected_component.iso_inv_image_comp_eq_iff_eq_map.mpr v'.prop⟩,
```